src/custom/components/AffiliateStatusCheck/index.tsx

```diff
@@ -83,7 +81,7 @@ export default function AffiliateStatusCheck() {
     }
 
     try {
-      await retry(() => uploadReferralDocAndSetDataHash(referralAddress.value), DEFAULT_RETRY_OPTIONS).promise
+      await retry(async () => uploadReferralDocAndSetDataHash(), DEFAULT_RETRY_OPTIONS).promise
```

Comment:
I think you might not need the retry here, as by this point the IPFS file has been uploaded already.
All that `uploadReferralDocAndSetDataHash` does now is to store the hash on redux.

No?

Reply:
Actually, what about reducing the scope of that hook and refactoring the code here a little bit.
Because now the upload is done in the hook before the callback is called, it might make sense to not use the callback anymore.

What do you think? You can ignore if this doesn't make sense

Reply:
@alfetopito I agree with you. If so, is it necessary to have a hook ? Since it won't be used to show data on the UI. It could be just a function. WDYT?

Reply:
The point of the hook is to use `swr` and delegate caching/whatever else to the lib
I mean, we want to make our lives easier. If it doesn't we can skip the lib